internal/database/lock.go

```diff
@@ -35,6 +36,59 @@ var (
 // UnlockFn can be deferred to release a lock.
 type UnlockFn func() error
 
+// MultiLock obtains multiple locks in a single transaction. Either all locks are obtained, or
+// the transaction is rolled back.
+// The lockIDs are sorted by normal ascending string sort order before obtaining the locks.
+func (db *DB) MultiLock(ctx context.Context, lockIDs []string, ttl time.Duration) (UnlockFn, error) {
```

Comment:
Is this so we can lock across multiple tables? It's unclear to me why we don't just use a transaction instead.

Reply:
added comments to worker.go where we take the locks